src/main/java/com/contentful/java/cda/AbsQuery.java

```diff
@@ -3,34 +3,419 @@
 import static com.contentful.java.cda.Util.resourcePath;
+import static java.lang.String.format;
 
-abstract class AbsQuery<T extends CDAResource, E extends AbsQuery<T, E>> {
-  final Class<T> type;
+/**
+ * Root of all queries.
+ * <p>
+ * This class includes options to query for entries, limit the amount of
+ * responses and more.
+ *
+ * @param <Resource> The type of the resource to be returned by this query
+ * @param <Query>    The query type to be returned on chaining to avoid casting on client side.
+ */
+abstract class AbsQuery<Resource extends CDAResource, Query extends AbsQuery<Resource, Query>> {
+  private static final String PARAMETER_CONTENT_TYPE = "content_type";
+  private static final String PARAMETER_SELECT = "select";
+  private static final String PARAMETER_ORDER = "order";
+  private static final String PARAMETER_LIMIT = "limit";
+  private static final String PARAMETER_SKIP = "skip";
+  private static final String PARAMETER_INCLUDE = "include";
+
+  final Class<Resource> type;
 
   final CDAClient client;
 
   final Map<String, String> params = new HashMap<String, String>();
 
-  AbsQuery(Class<T> type, CDAClient client) {
+  AbsQuery(Class<Resource> type, CDAClient client) {
     this.type = type;
     this.client = client;
   }
 
-  protected String path() {
-    return resourcePath(type);
+  /**
+   * Enforce this query to be using the parameter contentType.
+   * <p>
+   * The contentType is especially usefull if you want to limit the result of this query to only
+   * one type of contents. Also if you request that some fields should be respected by a filter
+   * operation, you have to make sure to set the content type with this methode, <b>before</b>
+   * applying the filter.
+   *
+   * @param contentType the content type to be used.
+   * @return the calling query for chaining.
+   * @throws IllegalArgumentException if contentType is null.
+   * @throws IllegalArgumentException if contentType is null.
+   * @throws IllegalStateException    if contentType was set before.
+   */
+  @SuppressWarnings("unchecked")
+  public Query withContentType(String contentType) {
+    checkNotEmpty(contentType, "contentType may not be empty.");
+
+    if (hasContentTypeSet()) {
+      throw new IllegalStateException(
+          format("ContentType \"%s\" is already present in query.", contentType)
+      );
+    } else {
+      params.put(PARAMETER_CONTENT_TYPE, contentType);
+    }
+
+    return (Query) this;
+  }
+
+  /**
+   * Limit response to only selected properties.
+   * <p>
+   * This will return an object, whose fields not specified with a select parameter will be set
+   * to null, resulting in potentially way smaller api response from the server.
+   * <p>
+   * As this SDK requires some introspection of the requested entries, it will always request the
+   * sys property. This also means, requesting a select with either 'sys' or 'sys.{someting}' will
+   * be already covered internally.
```

Comment:
Are you saying that regardless of the select operator, `sys` is always returned? If so, I'd suggest this change:

"`sys` properties are always returned despite any possible select operator usage, as the SDK uses `sys` internally"

I'm not as confident about this one though. Trust your gut ;-)